### PR TITLE
Feature/coordinated forks v5

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

**Review by dhedey — APPROVED** [review]
Have reviewed this across numerous other PRs, notably https://github.com/radixdlt/radixdlt/pull/656 - and have reviewed each individual commit since.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForkOverwritesFromPropertiesModule.java [hunk 1]
```diff
@@ -69,52 +69,67 @@
...
 
     @Override
-    public Set<ForkConfig> apply(Set<ForkConfig> forkConfigs) {
-      return forkConfigs.stream()
+    public Set<ForkBuilder> apply(Set<ForkBuilder> forkBuilders) {
+      return forkBuilders.stream()
           .map(
               c -> {
-                var epochOverwrite = properties.get("overwrite_forks." + c.name() + ".epoch", "");
+                final var forkDisabledOverwrite =
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForkOverwritesFromPropertiesModule.java:92]: I think this lambda worth moving into dedicated method and refactoring. Too much low-level logic in one method, IMO.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/NextCandidateForkPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  @Override
+  public LedgerAndBFTProof process(
+      LedgerAndBFTProof metadata,
+      EngineStore.EngineStoreInTransaction<LedgerAndBFTProof> engineStore,
+      List<REProcessedTxn> txns)
+      throws PostProcessorException {
+    if (metadata.getProof().getNextValidatorSet().isPresent()
+        && Forks.shouldCandidateForkBeEnacted(nextFork, metadata, forksEpochStoreSupplier.get())) {
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/NextCandidateForkPostProcessor.java:98]: I went with `isEpochChange(metadata) && shouldForkBeEnacted(metadata)`

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/NextFixedEpochForkPostProcessor.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  @Override
+  public LedgerAndBFTProof process(
+      LedgerAndBFTProof metadata,
+      EngineStore.EngineStoreInTransaction<LedgerAndBFTProof> engineStore,
+      List<REProcessedTxn> txns)
+      throws PostProcessorException {
+    if (metadata.getProof().getNextValidatorSet().isPresent()
+        && nextFork.epoch() == metadata.getProof().getEpoch() + 1) {
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/NextFixedEpochForkPostProcessor.java:92]: changed to `isEpochChange(metadata) && shouldNextForkBeEnacted(metadata)`

**Review by siy — APPROVED** [review]

**Pull request merged**